src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,125 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int MAX_CAPACITY = Integer.MAX_VALUE;
+    private static final int MIN_SIZE = 0;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = new Object[DEFAULT_CAPACITY];
+        size = MIN_SIZE;
+    }
+
+    private Object[] grow() {
```

Comment:
Do we actually need to return the array from it?

Let's try to return `void` and access `elementData` directly from here.